jenkins/opensearch/perf-test.jenkinsfile

```diff
@@ -0,0 +1,217 @@
+            trim: true
+        )
+        string(
+            defaultValue: '0',
+            name: 'WARMUP_ITERATIONS',
+            description: 'Number of times to run a workload before collecting data for Mensor (internal client).',
+            trim: true
+        )
+    }
+
+    stages {
+        stage('validate-parameters') {
+            steps {
+                script {
+                    if (BUNDLE_MANIFEST_URL == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Performance Tests failed to start. Missing parameter: BUNDLE_MANIFEST_URL.")
+                    }
+                    if (GITHUB_TOKEN == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Performance Tests failed to start. Missing parameter: GITHUB_TOKEN.")
+                    }
+                    if (TEST_ITERATIONS != null && !TEST_ITERATIONS.isInteger()) {
+                        currentBuild.result = 'ABORTED'
+                        error("Performance Tests failed to start. Invalid value for parameter: TEST_ITERATIONS. Value should be an integer.")
+                    }
+                    if (WARMUP_ITERATIONS != null && !WARMUP_ITERATIONS.isInteger()) {
+                        currentBuild.result = 'ABORTED'
+                        error("Performance Tests failed to start. Invalid value for parameter: WARMUP_ITERATIONS. Value should be an integer.")
+                    }
+                }
+            }
+        }
+        stage('perf-test') {
+            parallel {
+                stage('test-with-security') {
+                    agent {
+                        docker {
+                            label AGENT_LABEL
+                            image AGENT_IMAGE
+                            alwaysPull true
+                        }
+                    }
+                    steps {
+                        script {
+
+                            def bundleManifestObj = downloadBuildManifest(
+                                url: BUNDLE_MANIFEST_URL,
+                                path: BUNDLE_MANIFEST
+                            )
+                            String buildId = bundleManifestObj.getArtifactBuildId()
+                            env.BUILD_ID = buildId
+                            env.HAS_SECURITY = bundleManifestObj.components.containsKey("security")
+                            env.ARCHITECTURE = bundleManifestObj.getArtifactArchitecture()
+                            echo "BUNDLE_MANIFEST: ${BUNDLE_MANIFEST}"
+                            echo "BUILD_ID: ${BUILD_ID}"
+                            echo "BUILD_ID: ${BUILD_ID}"
+                            echo "Security: ${HAS_SECURITY}"
+                            echo "Architecture: ${ARCHITECTURE}"
+
+                            lib.jenkins.Messages.new(this).add('perf-test', "Performance tests for #${BUILD_ID}")
+                            if (env.HAS_SECURITY) {
```

Comment:
This stage will not do anything if Has_Security is false. Can we instead add it as a condition to the stage so that this stage is skipped instead?  For security false, this would run this stage with success which will be misleading.

Reply:
Updated the code to reflect this.